streams/src/test/java/org/apache/kafka/streams/StreamsBuilderTest.java

```diff
@@ -598,7 +598,7 @@ public void shouldUseSpecifiedNameForLeftJoinOperationBetweenKStreamAndKStream()
         final KStream<String, String> streamOne = builder.stream(STREAM_TOPIC);
         final KStream<String, String> streamTwo = builder.stream(STREAM_TOPIC_TWO);
 
-        streamOne.leftJoin(streamTwo, (value1, value2) -> value1, JoinWindows.of(Duration.ofHours(1)), Joined.as(STREAM_OPERATION_NAME));
+        streamOne.leftJoin(streamTwo, (value1, value2) -> value1, JoinWindows.of(Duration.ofHours(1)), Joined.as(STREAM_OPERATION_NAME), Materialized.as(STREAM_OPERATION_NAME));
```

Comment:
In this test, breaking the naming of the store from `Joined` results in test failures, as the expected name of the store has changed.  Instead of updating the expected topology, I pass in the name of the store.  

IMHO it's better to have the code produce the correct expected topology vs. changed the expected topology strings.

Same for other tests in `StreamsBuilderTest`.

Reply:
Should we have two tests (one for each case)?